source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -279,36 +290,31 @@ namespace aspect
       }
 
 
-
       template <int dim>
       std::pair<double, double>
-      CompositeViscoPlastic<dim>::compute_strain_rate_and_derivative (const double creep_stress,
-                                                                      const double pressure,
-                                                                      const double temperature,
-                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
-                                                                      const DislocationCreepParameters dislocation_creep_parameters,
-                                                                      const PeierlsCreepParameters peierls_creep_parameters,
-                                                                      const DruckerPragerParameters drucker_prager_parameters) const
+      CompositeViscoPlastic<dim>::calculate_log_strain_rate_and_deriv(const std::vector<std::pair<double, double>> &log_edot_and_deriv) const
       {
-        std::pair<double, double> creep_edot_and_deriv = std::make_pair(0., 0.);
+        // dlneps_dlnsigma = sum_i(stress_exponent_i * edot_i) / sum_i(edot_i)
+        // delta_lneps = log(sum_i(edot_i)) - log(edot)
```

Comment:
These comments need to be attached to the correct variables and require some more text. After some minutes of checking the code and the names I think I figured out that `dlneps_dlnsigma` (i.e. the logarithm of stress derivative of the logarithm of strain rate) is computed as the average of the stress derivatives of the individual flow mechanisms weighted by their contribution to the total strain rate (I think?). But I do not see in the code how the stress_exponent_i in the comment plays into this. I would rework this code the following way: Add comments here, that just explain what `edot` and `sumedotn` are, maybe both could also get simpler names (`strain_rate_sum` and `weighted_stress_derivative_sum`?). Then move the formulas how the return variables are computed down to the `return` statement.